### PR TITLE
Implement ramalama run/serve

### PR DESCRIPTION
**Pull request by ericcurtin**
Now we can run "ramalama run/serve granite-code", if not using
container, one must at least build/install llama.cpp. Added huggingface
support.

### CODE REVIEW

File: ramalama [hunk 1]
```diff
@@ -68,9 +68,9 @@ def run_curl_command(args, filename):
             sys.exit(e.returncode)
 
 
-def pull_ollama_manifest(ramalama_store, manifests, accept, registry_head, model_tag):
+def pull_ollama_manifest(repos_ollama, manifests, accept, registry_head, model_tag):
```
Comment (rhatdan) [inline ramalama:71]: Why not just call it repos.  No need to keep repeating _ollama

File: ramalama [hunk 2]
```diff
@@ -183,14 +202,26 @@ def select_container_manager():
...
+
     if len(sys.argv) < 2:
         usage()
 
-    ramalama_store = get_ramalama_store()
     command = sys.argv[1]
     if command == "pull" and len(sys.argv) > 2:
-        pull_cli(ramalama_store + "/repos/ollama",
-                 ramalama_store + "/models/ollama", sys.argv[2])
+        pull_cli(ramalama_store, sys.argv[2])
```
Comment (rhatdan) [inline ramalama:220]: switch command {
case "pull":
case "run":
}

Comment (rhatdan) [reply ramalama:220]: Do you have commands that handle 2 options?

File: ramalama (hunk 1 above)
Comment (ericcurtin) [reply ramalama:71]: My thinking was shortly I will bring in hugging face support again and there will likely be "repos_hf", "repos_ollama" directory is:

~/.local/share/ramalama/repos/ollama

shortly there will be:

~/.local/share/ramalama/repos/huggingface

File: ramalama (hunk 2 above)
Comment (ericcurtin) [reply ramalama:220]: Yeah there will be:

ramalama list/ls

as an example that lists all models downloaded and available is coming.

Will change to switch, apparently they call it match in python for some reason

File: ramalama (hunk 1 above)
Comment (rhatdan) [reply ramalama:71]: I think you should embed the location of the store inside of the functions, and not force it to be different.

Once we have multiple pull_manifests() functions, then we can start to consildate functionality and get as much reuse as possible.

File: ramalama (hunk 2 above)
Comment (ericcurtin) [reply ramalama]: How would we feel about going back to if/else if/else? The python switch equivalent in Python 3.10 (2021) and macOS build is failing here, I'm guessing the python3 version on macOS is older

Comment (ericcurtin) [reply ramalama]: Going back to if, else, etc. fixed it anyway

**Comment by rhatdan** [comment]
Ok merging, we can cleanup later.

**Pull request merged**